src/overlays/actors/ovl_Bg_Mori_Elevator/z_bg_mori_elevator.c

```diff
@@ -20,37 +34,247 @@ const ActorInit Bg_Mori_Elevator_InitVars = {
+
+        if (target < *posY) {
+            *posY = target;
+        }
+    } else {
+        if (target < *posY) {
+            if (phi_f2 < (-yVel)) {
+                phi_f2 = (-yVel);
+            } else {
+                if ((-arg4) < phi_f2) {
+                    phi_f2 = (-arg4);
+                }
+            }
+            *posY = (*posY + phi_f2);
+            if (*posY < target) {
+                *posY = target;
+            }
+        } else {
+            phi_f2 = 0.0f;
+        }
+    }
+    return phi_f2;
+}
+
+void func_808A18FC(BgMoriElevator* this, f32 distTo) {
+    f32 phi_f12;
+    f32 temp_f2;
+    f32 temp1;
+
+
+    temp_f2 = fabsf(distTo) * 0.09f; 
+    if (temp_f2 < 0.0f) {
+    // clang-format off
+        phi_f12 = 0.0f; } else { temp1 = (temp_f2 > 1.0f) ? (1.0f) : (temp_f2);phi_f12 = temp1;} func_800F436C(&this->dyna.actor.projectedPos, NA_SE_EV_ELEVATOR_MOVE2 - SFX_FLAG, phi_f12);
+    // clang-format on
+}
+
+void BgMoriElevator_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgMoriElevator* this = THIS;
+    s32 pad;
+    s32 sp24;
+    sp24 = 0;
+
+    this->unk_172 = sIsSpawned;
+    this->objectIndex = Object_GetIndex(&globalCtx->objectCtx, OBJECT_MORI_TEX);
+    if (this->objectIndex < 0) {
+        Actor_Kill(thisx);
+        // Forest Temple obj elevator Bank Danger!
+        osSyncPrintf("Error : 森の神殿 obj elevator バンク危険！(%s %d)\n", "../z_bg_mori_elevator.c", 277);
+    } else {
+        switch (sIsSpawned) {
+            case false:
+                // Forest Temple elevator CT
+                osSyncPrintf("森の神殿 elevator CT\n");
+                sIsSpawned = true;
+                this->dyna.actor.room = -1;
+                Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+                DynaPolyInfo_SetActorMove(&this->dyna, DPM_PLAYER);
+                DynaPolyInfo_Alloc(&D_060035F8, &sp24);
+                this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, thisx, sp24);
+                BgMoriElevator_SetupPlaceInGround(this);
+                break;
+            case true:
+                Actor_Kill(thisx);
+                break;
+        }
+    }
+}
+
+void BgMoriElevator_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgMoriElevator* this = THIS;
+
+    if (this->unk_172 == 0) {
+        // Forest Temple elevator DT
+        osSyncPrintf("森の神殿 elevator DT\n");
+        DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+        sIsSpawned = false;
+    }
+}
+
+s32 BgMoriElevator_IsLinkRiding(BgMoriElevator* this, GlobalContext* globalCtx) {
+    return ((this->dyna.unk_160 & 2) && !(this->isRiding & 2) &&
+            ((PLAYER->actor.posRot.pos.y - this->dyna.actor.posRot.pos.y) < 80.0f));
+}
+
+void BgMoriElevator_SetupPlaceInGround(BgMoriElevator* this) {
+    this->actionFunc = BgMoriElevator_PlaceInGround;
+}
+
+void BgMoriElevator_PlaceInGround(BgMoriElevator* this, GlobalContext* globalCtx) {
```

Comment:
Considering what this action does, I suggest something like `WaitOnInit` or `WaitAfterInit`

Reply:
I like the name it has now since it explicitly states what it is doing.  Maybe `WaitInGround`?

Reply:
This action is meant specifically to wait for the secondary object to load after init, so the name should reflect that imo. Also I don't see anything that makes it be in ground during this action, so that wouldn't be very accurate for modding purposes imo.